# HG changeset patch
# User Benjamin Lindner <lindnerb@users.sourceforge.net>
# Date 1243068028 -7200
#      Sat May 23 10:40:28 2009 +0200
# Branch release-3-2-x
# Node ID 015af716bdd61eae0ca073ed480e25cd0255a17c
# Parent  a80076bea5bda579bff75a32b816b1dd9106fce7
properly #ifdef MSVC linker specific flags

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2009-05-22 Benjamin Lindner <lindnerb@users.sourceforge.net>
+
+	* mkoctfile.cc.in: mask MSVC specific linker flags for mex output
+
 	2009-05-25  Jaroslav Hajek  <highegg@gmail.com>
 
 	Version 3.2.0 released.
 
 2009-06-02  Robert T. Short  <octave@phaselockedsystems.com>
 
 	* examples/@polynomial: Fix end method for zero based indexing.
 
diff --git a/mkoctfile.cc.in b/mkoctfile.cc.in
--- a/mkoctfile.cc.in
+++ b/mkoctfile.cc.in
@@ -528,17 +528,19 @@ main (int argc, char **argv)
 	}
       else if (arg == "--link-stand-alone")
 	{
 	  link_stand_alone = true;
 	}
       else if (arg == "--mex")
 	{
 	  incflags += " -I.";
+#ifdef _MSC_VER
 	  ldflags += " -Wl,-export:mexFunction";
+#endif
 	  output_ext = ".mex";
 	}
       else if (starts_with (arg, "-W"))
 	{
 	  pass_on_options += (" " + arg);
 	}
       else
 	{
